polina.vasileva/task-2-2/pkg/intheap/intheap.go

```diff
@@ -0,0 +1,48 @@
+package intheap
+
+import "fmt"
+
+type IntHeap []int
+
+func (h *IntHeap) Len() int {
+	return len(*h)
+}
+
+func (h *IntHeap) Less(i, j int) bool {
+	if i >= h.Len() || j >= h.Len() {
+		panic("Index out of range in intheap")
+	}
+
+	return (*h)[i] > (*h)[j]
+}
+
+func (h *IntHeap) Swap(i, j int) {
+	if i >= h.Len() || j >= h.Len() {
+		panic("Index out of range in intheap")
+	}
+
+	(*h)[i], (*h)[j] = (*h)[j], (*h)[i]
+}
+
+func (h *IntHeap) Push(x interface{}) {
+	value, ok := x.(int)
+	if !ok {
+		panic(fmt.Sprintf("IntHeap.Push: expected int, got %T", x))
```

Comment:
```suggestion
		panic(fmt.Sprintf("expected int, got %T", x))
```

оно тут лишнее, это будет очевидно из стека вызовов